File: JavaScript/6js 2/toCamelCase.js
/*
|--------------------------------------------------------------------------
| Convert string to camel case
|--------------------------------------------------------------------------
|
*/

function toCamelCase(str) {
   return str.replace(/[-_](.)/g, function (a, b) {
      return b.toUpperCase();
   });
}


function toCamelCase(str) {
   return str.replace(/[_-]\w/gi, (ch) => ch[1].toUpperCase());
}


function toCamelCase(str) {
   return str
      .split(/-|_/g)
      .map((w, i) => (i > 0 ? w.charAt(0).toUpperCase() : w.charAt(0)) + w.slice(1))
      .join("");
}


console.log(toCamelCase(""), "");
console.log(toCamelCase("the_stealth_warrior"), "theStealthWarrior");
console.log(toCamelCase("The-Stealth-Warrior"), "TheStealthWarrior");
console.log(toCamelCase("A-B-C"), "ABC");

File: Python/6py 2/to_camel_case.py
"""
|--------------------------------------------------------------------------
| Convert string to camel case
|--------------------------------------------------------------------------
|
"""

from re import sub
def to_camel_case(text):
   return sub(r'[_-]\w', lambda x: x.group(0)[1].upper(), text)


def to_camel_case(text):
   return text[:1] + text.title()[1:].replace('_', '').replace('-', '')


def to_camel_case(text):
   words = text.replace('_', '-').split('-')
   return words[0] + ''.join([x.title() for x in words[1:]])


print(to_camel_case(""), "")
print(to_camel_case("the_stealth_warrior"), "theStealthWarrior")
print(to_camel_case("The-Stealth-Warrior"), "TheStealthWarrior")
print(to_camel_case("A-B-C"), "ABC")